controllers/userController.js

```diff
@@ -0,0 +1,173 @@
+    }
+
+    const avatarURL = req.file
+      ? `/avatars/${req.file.filename}`
+      : gravatar.url(email, { s: "250", d: "retro" }, true);
+
+    const newUser = new User({ email, avatarURL });
+    newUser.setPassword(password);
+    await newUser.save();
+
+    const token = jwt.sign({ id: newUser._id }, process.env.JWT_SECRET);
+
+    res.status(201).json({
+      status: "success",
+      code: 201,
+      data: {
+        user: {
+          email: newUser.email,
+          subscription: newUser.subscription,
+          avatarURL: newUser.avatarURL,
+        },
+        token,
+      },
+    });
+  } catch (error) {
+    console.error(error);
+    res
+      .status(500)
+      .json({ status: "error", code: 500, message: "Internal server error" });
+  }
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  try {
+    const user = await User.findOne({ email });
+
+    if (!user || !user.validPassword(password)) {
+      return res.status(401).json({
+        status: "error",
+        code: 401,
+        message: "Email or password is wrong",
+      });
+    }
+
+    const token = jwt.sign({ id: user._id }, process.env.JWT_SECRET, {
+      expiresIn: "5d",
+    });
+
+    user.token = token;
+    await user.save();
+
+    res.status(200).json({
+      status: "success",
+      code: 200,
+      data: {
+        token,
+        user: {
+          email: user.email,
+          subscription: user.subscription,
+          avatarURL: user.avatarURL,
+        },
+      },
+    });
+  } catch (error) {
+    console.error(error);
+    res
+      .status(500)
+      .json({ status: "error", code: 500, message: "Internal server error" });
+  }
+};
+
+const logout = async (req, res) => {
+  try {
+    req.user.token = null;
+    await req.user.save();
+    res
+      .status(204)
+      .json({ status: "success", code: 204, message: "Logout successful" });
+  } catch (error) {
+    console.error(error);
+    res
+      .status(500)
+      .json({ status: "error", code: 500, message: "Internal server error" });
+  }
+};
+
+const getCurrentUser = async (req, res) => {
+  try {
+    const { email, subscription, avatarURL } = req.user;
+    res.status(200).json({
+      status: "success",
+      code: 200,
+      data: {
+        email,
+        subscription,
+        avatarURL,
+      },
+    });
+  } catch (error) {
+    console.error(error);
+    res
+      .status(500)
+      .json({ status: "error", code: 500, message: "Internal server error" });
+  }
+};
+
+const updateAvatar = async (req, res) => {
+  try {
+    if (!req.file) {
+      return res
+        .status(400)
+        .json({ status: "error", code: 400, message: "File is required" });
+    }
+
+    const { id } = req.user;
+    const { path: temporaryPath, filename } = req.file;
+    const fileExtension = path.extname(filename);
+
+    const allowedExtensions = [".jpg", ".jpeg", ".png"];
```

Comment:
To sprawdzasz w pliku upload.js a jeżeli nie działa to poprawnie to co tam zrobiłaś to trzeba poprawić tak abyś tutaj nie musiała już robić walidacji.